lib/index.js

```diff
@@ -847,16 +847,15 @@ class Image extends RustClass {
 
   get src(){ return this.prop('src') }
   set src(src){
-    var noop = () => {},
-        onload = img => fetch.emit('ok', img),
+    var onload = img => fetch.emit('ok', img),
         onerror = err => fetch.emit('err', err),
-        passthrough = fn => arg => { (fn||noop)(arg); delete this._fetch },
+        passthrough = fn => arg => { fn && fn(arg); delete this._fetch },
         data
 
     if (this._fetch) this._fetch.removeAllListeners()
     let fetch = this._fetch = new EventEmitter()
-        .once('ok', passthrough(this.onload))
-        .once('err', passthrough(this.onerror))
+        .once('ok', passthrough(this.onload.bind(this)))
+        .once('err', passthrough(this.onerror.bind(this)))
```

Comment:
These will throw if the callbacks haven't been set.  
Just need a null check.

`this.onload?.bind(this)`